components/segment_images/src/main.py

```diff
@@ -109,32 +125,22 @@ def transform(
         device = "cuda" if torch.cuda.is_available() else "cpu"
         logger.info(f"Device: {device}")
 
-        processor = AutoImageProcessor.from_pretrained(model_id)
+        processor = SegformerImageProcessor.from_pretrained(model_id)
```

Comment:
Based on [this](https://github.com/huggingface/transformers/blob/28aa438cd2759cfdf1da9326ea6cde4496d1be23/src/transformers/models/auto/image_processing_auto.py#L97), I think UperNet does use the `SegformerImageProcessor`? But I'm not as familiar with this as you :sweat_smile: 

Using the `AutoImageProcessor` is nice in theory, but since the different ImageProcessors don't implement the same interface, I actually think this component only supports models using the `SegformerImageProcessor` anyway? Also for typing, it helps to know which `Processor` will be selected.

Reply:
Indeed would be nice to have a generic component that can leverage any HF segmentation model and users can change the model based on a model ID. Makes them more re-usable. But if the models have different interfaces that won't be possible